src/java/org/apache/cassandra/index/sai/iterators/KeyRangeIterator.java

```diff
@@ -106,13 +107,21 @@ else if (next.compareTo(nextKey) >= 0)
         return recomputeNext();
     }
 
+    /**
+     * Skip to nextKey.
+     *
+     * That is, implementations should set up the iterator state such that
+     * calling computeNext() will return nextKey if present,
+     * or the first one after it if not present.
+     */
     protected abstract void performSkipTo(PrimaryKey nextKey);
 
-    protected PrimaryKey recomputeNext()
+    private PrimaryKey recomputeNext()
     {
         return tryToComputeNext() ? peek() : endOfData();
     }
 
+    // protected because inherited from Guava. We don't want to expose this method.
```

Comment:
We should probably annotate the method with `@Override`.

Reply:
done